src/jaeger_passage_reporter_http.erl

```diff
@@ -0,0 +1,173 @@
+
+-record(?STATE,
+        {
+          endpoint             :: string(),
+          options              :: start_options(),
+          http_client          :: http_client(),
+          default_service_name :: atom(),
+          process_tags         :: passage:tags()
+        }).
+
+-define(CONTENT_TYPE, {"Content-Type", "application/x-thrift"}).
+
+%%------------------------------------------------------------------------------
+%% Exported Types
+%%------------------------------------------------------------------------------
+
+-type start_options() :: [start_option()].
+%% Options for {@link start/2}.
+
+-type start_option() :: {endpoint, string()}
+                      | {http_client, http_client()}
+                      | {default_service_name, atom()}
+                      | {process_tags, passage:tags()}.
+%% <ul>
+%%   <li><b>endpoint</b>: The jaeger endpoint URL for sending thrift messages. The default value is `http://127.0.0.1:14268'.</li>
+%%   <li><b>http_client</b>: The callback to call to send span to jaeger.</li>
+%%   <li><b>default_service_name</b>: The default service name. If a reporting span has `location.application' tag, the value is used as the service name instead of this. The default value is `ReporterId'.</li>
+%%   <li><b>process_tags</b>: The tags of the reporting process. The default value is `#{}'.</li>
+%% </ul>
+%% Example of a http_client calback
+%% Client = fun(Url, Method, Headers, Body, ReporterOptions) ->
+%%    User = proplists:get_value(user, ReporterOptions),
+%%    Password = proplists:get_value(password, ReporterOptions),
+%%    ibrowse:send_req(Url, Headers, Method, Body, [{basic_auth, {User,  Password}}])
+%% end.
+
+
+-type http_client() :: fun((
+        Url    :: string(),
+        Method :: post,
+        Headers :: [{string(), string()}],
+        Body :: string() | binary(),
+        ReporterOptions :: start_options()) ->
+    ok).
+
+%%------------------------------------------------------------------------------
+%% Application Internal Functions
+%%------------------------------------------------------------------------------
+%% @private
+-spec start_link(jaeger_passage_reporter:reporter_id(), start_options()) -> {ok, pid()} | {error, Reason} when
+      Reason :: {already_started, pid()} | term().
+start_link(ReporterId, Options) ->
+    Name = jaeger_passage_local_ns:reporter_name(ReporterId),
+    gen_server:start_link(Name, ?MODULE, {ReporterId, Options}, []).
+
+%%------------------------------------------------------------------------------
+%% 'gen_server' Callback Functions
+%%------------------------------------------------------------------------------
+%% @private
+init({ReporterId, Options}) ->
+    Endpoint = proplists:get_value(endpoint, Options, "http://127.0.0.1:14268"),
+    EndpointURL = Endpoint ++ "/api/traces",
+
+    HttpClient = proplists:get_value(http_client, Options),
+    HttpClient =/= undefined orelse error(badarg, [ReporterId, Options]),
```

Comment:
This is just an idea and can be ignored, but how about using [`httpc`](http://erlang.org/doc/man/httpc.html) as the default HTTP client?

Reply:
If we would use `httpc` we would have to include it into `rebar.config`. Otherwise we would be calling undefined function and xref check would fail. This wouldn't work for users who want to use different client.

Reply:
Hmm, if my understanding is correct, it woundn't need to modify `rebar.config` when we use `httpc` since it's a built-in standard module (http://erlang.org/doc/man/httpc.html).
However, I'm not 100% sure, so please let me know if I'm wrong.

Reply:
added in https://github.com/sile/jaeger_passage/pull/3/commits/dd36ad40df5e198bda39901e1779ed1e9f51e775

Reply:
Thanks!